### PR TITLE
Merge Data.Parser.ParserD test cases to Data.Parser

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Review by harendra-kumar — COMMENTED** [review]
Parser was ParserK earlier, and ParserD was ParserD. So we still need ParserK tests, we cannot just remove it, but have to adapt to ParserK. 

For now we can just use Test.Data.Parser as it is for Data.Parser tests and rename Test.Data.Parser.ParserD to Test.Data.ParserK . And in a later PR adapt this to ParserK.

File: test/Streamly/Test/Data/Parser.hs [hunk 1]
```diff
@@ -54,17 +59,18 @@ max_value = 10000
...
 fromFold :: Property
 fromFold =
-    forAll (listOf $ chooseInt (min_value, max_value)) $ \ls ->
-        monadicIO $ do
-        s1 <- S.parse (P.fromFold FL.sum) (S.fromList ls)
-        o2 <- S.fold FL.sum (S.fromList ls)
-        return $
-            case s1 of
-                Right o1 -> o1 == o2
+    forAll (listOf $ chooseInt (min_value, max_value))
```
Comment (adithyaov) [inline test/Streamly/Test/Data/Parser.hs:69]: Keep the earlier implementation of `fromFold` . It's easier to read.

File: test/Streamly/Test/Data/Parser.hs [hunk 2]
```diff
@@ -161,68 +158,59 @@ satisfy =
...
-                forAll (vectorOf list_length (chooseInt (min_value, max_value)))
-                    $ \ls ->
-                        case runIdentity $ S.parse (P.takeBetween m n FL.toList)
-                                (S.fromList ls) of
-                            Right parsed_list ->
-                                let lpl = Prelude.length parsed_list
-                                in checkListEqual parsed_list
-                                    $ Prelude.take lpl ls
-                            Left _ -> property False
+                forAll (vectorOf list_length (chooseInt (min_value, max_value))) $ \ls ->
```
Comment (adithyaov) [inline test/Streamly/Test/Data/Parser.hs:167]: Keep the earlier implementation of `takeBetweenPass`

File: test/Streamly/Test/Data/Parser.hs [hunk 3]
```diff
@@ -234,20 +222,20 @@ takeEQ =
             in
                 case runIdentity $ S.parse (P.takeEQ n FL.toList) (S.fromList ls) of
                     Right parsed_list ->
-                        if n <= list_length
-                        then checkListEqual parsed_list (Prelude.take n ls)
-                        else property False
+                        if (n <= list_length) then
```
Comment (adithyaov) [inline test/Streamly/Test/Data/Parser.hs:225]: The earlier indentation is better. Keep the earlier diff here.

File: test/Streamly/Test/Data/Parser.hs [hunk 4]
```diff
@@ -258,12 +246,12 @@ takeGE =
             in
                 case runIdentity $ S.parse (P.takeGE n FL.toList) (S.fromList ls) of
                     Right parsed_list ->
-                        if n <= list_length
-                        then checkListEqual parsed_list ls
-                        else property False
+                        if (n <= list_length) then
```
Comment (adithyaov) [inline test/Streamly/Test/Data/Parser.hs:249]: Keep the earlier diff.

**Comment by adithyaov** [comment]
Some comments. Looks good overall.

File: streamly.cabal [hunk 5]
```diff
@@ -107,7 +107,7 @@ extra-source-files:
     test/Streamly/Test/Data/Array/Mut.hs
     test/Streamly/Test/Data/Ring/Unboxed.hs
     test/Streamly/Test/Data/Array/Stream.hs
-    test/Streamly/Test/Data/Parser/ParserD.hs
+    test/Streamly/Test/Data/ParserK.hs
```
Comment (harendra-kumar) [inline streamly.cabal:110]: We are not adding test/Streamly/Test/Data/Parser here?

**Pull request merged**